### ISSUE TITLE
Force clickhouse connections to be read-only by default

### ISSUE THREAD

**Issue by hwchen** [tesseract-olap/tesseract#205]
For any clickhouse connection in tesseract, check if `readonly` parameter is set. If it is not explicitly set, then it should be added to the list of connection parameters as `?readonly=1` or `&readonly=1` (if there are other parameters)

**Comment by MarcioPorto** [comment]
Another way of doing this, which I have added to all projects is to boot Tesseract with a user that only has read permissions. @jspeis do you still want to explore this further or should I close this issue?

**Comment by jspeis** [comment]
I still like this idea as an added layer of protection when people are spinning up new projects.

**Comment by MarcioPorto** [comment]
@jspeis Will this still be necessary once #241 is merged?

**Comment by jspeis** [comment]
as of clickhouse-rs v0.1.19 the driver now [supports](https://github.com/suharev7/clickhouse-rs/blob/master/src/lib.rs#L37) adding `?readonly=1` to the connection string. So could be nice to have this be a default option for clickhouse connections
